### PR TITLE
Combine the social auth info in a single place

### PR DESCRIPTION
**Pull request by yurynix**
This fixes #16638 not properly rebased after #16955

The main change is in `client/state/login/reducer.js`, all others changes are derived from that change.

#### Testing

1. Run `git checkout update/social-linking` and start your server, or open a [live branch](https://calypso.live/?branch=update/social-linking)
2. Open the [`start/social` page](http://calypso.localhost:3000/start/social)
3. Check that "signup" with existing google account takes to social connect and you are able to connect the account and the account is actually linked
4. Check that in login [flow](http://calypso.localhost:3000/log-in) you can still link a social account to existing WPCOM account and it is actually being linked

#### Reviews
  
- [x] Code
- [x] Product
- [x] Tests

### CODE REVIEW

File: client/state/login/reducer.js [hunk 1]
```diff
@@ -174,29 +174,29 @@ export const socialAccount = createReducer( { isCreating: false, createError: nu
...
-	[ SOCIAL_CREATE_ACCOUNT_REQUEST_FAILURE ]: ( state, { error, token, service } ) => {
-		if ( error.code === 'user_exists' ) {
-			return {
-				isLinking: true,
-				email: error.email,
-				token,
-				service,
-			};
-		}
+const userExistsErrorHandler = ( state, { error, authInfo } ) => {
```
Comment (Tug) [inline client/state/login/reducer.js]: I find a bit confusing that we have to pass `authInfo` for both `SOCIAL_CREATE_ACCOUNT_REQUEST_FAILURE ` and `SOCIAL_LOGIN_REQUEST_FAILURE`. Couldn't we just get it from the state? (of course `SOCIAL_CREATE_ACCOUNT_REQUEST` and `SOCIAL_LOGIN_REQUEST` would need to store it there first).

Comment (yurynix) [inline client/state/login/reducer.js]: Not all calls to the API are within an action creator ( [example](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/step-actions.js#L295) ), so I don't think we should rely on `socialInfo` being available on the store and create a dependency to it being available in some previous step.

Comment (Tug) [inline client/state/login/reducer.js]: Right, I guess that makes sense!

File: client/state/login/selectors.js [hunk 2]
```diff
@@ -218,28 +218,20 @@ export const getCreateSocialAccountError = ( state ) => get( state, 'login.socia
 export const getRequestSocialAccountError = ( state ) => get( state, 'login.socialAccount.requestError', null );
 
 /***
- * Gets the email address of the social account to be linked.
+ * Gets the OAuth2 client data.
```
Comment (scruffian) [inline client/state/login/selectors.js]: I think this all lives in `client/state/login/oauth` now.

File: client/blocks/login/index.jsx [hunk 3]
```diff
@@ -250,9 +250,9 @@ export default connect(
 		requestNotice: getRequestNotice( state ),
 		twoFactorEnabled: isTwoFactorEnabled( state ),
 		twoFactorNotificationSent: getTwoFactorNotificationSent( state ),
-		linkingSocialUser: getLinkingSocialUser( state ),
-		linkingSocialService: getLinkingSocialService( state ),
 		oauth2ClientData: getOAuth2ClientData( state ),
+		isLinking: getSocialAccountIsLinking( state ),
```
Comment (Tug) [inline client/blocks/login/index.jsx]: I don't see where this function is defined

File: client/state/login/selectors.js [hunk 4]
```diff
@@ -263,4 +239,12 @@ export const getSocialAccountLinkEmail = ( state ) => get( state, 'login.socialA
  * @param  {Object}   state  Global state tree
```
Comment (Tug) [inline client/state/login/selectors.js]: Oh right it was hidden in there, sorry

**Pull request merged**